File: app/src/main/java/br/com/gsas/app/picpay/Database/AppDatabase.java
package br.com.gsas.app.picpay.Database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import br.com.gsas.app.picpay.DAO.CartaoDAO;
import br.com.gsas.app.picpay.DAO.FeedDAO;
import br.com.gsas.app.picpay.Domain.Cartao;
import br.com.gsas.app.picpay.Domain.Feed;
import br.com.gsas.app.picpay.Util.ConverterContato;

@Database(entities = {Cartao.class, Feed.class}, version = 1, exportSchema = false)
@TypeConverters({ConverterContato.class})
public abstract class AppDatabase extends RoomDatabase {

    public abstract CartaoDAO cartaoDAO();

    public abstract FeedDAO feedDAO();

}

File: app/src/main/java/com/michaeljordan/testemobilepicpay/data/local/AppDataBase.kt
package com.michaeljordan.testemobilepicpay.data.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import com.michaeljordan.testemobilepicpay.data.local.dao.CardDao
import com.michaeljordan.testemobilepicpay.model.Card

@Database(entities = arrayOf(Card::class), version = 1, exportSchema = false)
abstract class AppDataBase : RoomDatabase() {

    abstract fun cardDao(): CardDao

    companion object {
        private val DATABASE_NAME = "testepicpay"
        private lateinit var instance: AppDataBase

        fun getAppDatabase(context: Context): AppDataBase {
            synchronized(AppDataBase::class.java) {
                instance = Room.databaseBuilder(
                    context.applicationContext,
                    AppDataBase::class.java,
                    DATABASE_NAME
                )
                    .build()
            }
            return instance
        }
    }
}
